sdk-java/src/main/java/io/littlehorse/sdk/wfsdk/internal/WaitForThreadsNodeOutputImpl.java

```diff
@@ -15,13 +15,45 @@ public WaitForThreadsNodeOutputImpl(String nodeName, WorkflowThreadImpl parent,
         this.threadSpec = threadSpec;
     }
 
-    public WaitForThreadsNodeOutputImpl withPolicy(WaitForThreadsPolicy failureStrategy) {
-        Node nodesOrThrow = threadSpec.getNodesOrThrow(nodeName);
-        WaitForThreadsNode waitForThreads = nodesOrThrow.getWaitForThreads();
-        waitForThreads = waitForThreads.toBuilder().setPolicy(failureStrategy).build();
-        nodesOrThrow =
-                nodesOrThrow.toBuilder().setWaitForThreads(waitForThreads).build();
-        threadSpec.putNodes(nodeName, nodesOrThrow);
+    public WaitForThreadsNodeOutput handleExceptionOnChild(String exceptionName, ThreadFunc handler) {
```

Comment:
In my opinion, we should not allow the exception name Null here because we set FailureHandlerDef.LHFailureType.FAILURE_TYPE_ERROR when the exception name is null.

Reply:
Good catch, I am working on unit tests and hopefully would have caught this. The correct answer is that it should be `FAILURE_TYPE_EXCEPTION` if `exceptionName == null`, not `FAILURE_TYPE_ERROR`.